components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/DefaultKeyValidationHandler.java

```diff
@@ -170,32 +180,69 @@ public boolean validateScopes(TokenValidationContext validationContext) throws A
                 + ":" +
                 validationContext.getHttpVerb();
 
-        boolean isValid = false;
+        Set<OAuth2ScopeValidator> oAuth2ScopeValidators = OAuthServerConfiguration.getInstance()
+                .getOAuth2ScopeValidators();
+        //validate scope for filtered validators from db
+        String[] scopeValidators;
+        OAuthAppDO appInfo;
```

Comment:
Can we move variable definition to try block? I don't see a usage of it outside the try block.

Reply:
done